probe-rs-cli-util/src/common_options.rs

```diff
@@ -0,0 +1,473 @@
+        long = "elf",
+        help = "The path to the ELF file to be flashed."
+    )]
+    pub elf: Option<String>,
+    #[structopt(
+        name = "directory",
+        long = "work-dir",
+        help = "The work directory from which cargo-flash should operate from."
+    )]
+    pub work_dir: Option<String>,
+    #[structopt(flatten)]
+    /// Arguments which are forwarded to 'cargo build'.
+    pub cargo_options: CargoOptions,
+    #[structopt(flatten)]
+    /// Argument relating to probe/chip selection/configuration.
+    pub probe_options: ProbeOptions,
+}
+
+impl FlashOptions {
+    pub fn early_exit(&self, f: impl Write) -> Result<bool, OperationError> {
+        if self.list_probes {
+            list_connected_probes(f)?;
+            return Ok(true);
+        }
+
+        if self.list_chips {
+            print_families(f)?;
+            return Ok(true);
+        }
+
+        Ok(false)
+    }
+}
+
+#[derive(StructOpt, Debug)]
+pub struct ProbeOptions {
+    #[structopt(name = "chip", long = "chip")]
+    pub chip: Option<String>,
+    #[structopt(name = "chip description file path", long = "chip-description-path")]
+    pub chip_description_path: Option<String>,
+    #[structopt(name = "protocol", long = "protocol", default_value = "swd")]
+    pub protocol: WireProtocol,
+    #[structopt(
+        long = "probe",
+        help = "Use this flag to select a specific probe in the list.\n\
+        Use '--probe VID:PID' or '--probe VID:PID:Serial' if you have more than one probe with the same VID:PID."
+    )]
+    pub probe_selector: Option<DebugProbeSelector>,
+    #[structopt(
+        long = "connect-under-reset",
+        help = "Use this flag to assert the nreset & ntrst pins during attaching the probe to the chip."
+    )]
+    pub connect_under_reset: bool,
+    #[structopt(name = "speed", long = "speed", help = "The protocol speed in kHz.")]
+    pub speed: Option<u32>,
+    #[structopt(long = "dry-run")]
+    pub dry_run: bool,
+}
+
+static mut PROBE: Option<Probe> = None;
+static mut TARGET_SELECTOR: Option<TargetSelector> = None;
+
+impl ProbeOptions {
+    /// Add targets contained in file given by --chip-description-path
+    /// to probe-rs registery.
+    pub fn maybe_load_chip_desc(&self) -> Result<(), OperationError> {
+        if let Some(ref cdp) = self.chip_description_path {
+            probe_rs::config::add_target_from_yaml(&Path::new(cdp)).map_err(|error| {
+                OperationError::FailedChipDescriptionParsing {
+                    source: error,
+                    path: cdp.clone(),
+                }
+            })
+        } else {
+            Ok(())
+        }
+    }
+
+    pub fn attach_to_probe(&self) -> Result<&Probe, OperationError> {
+        // Tries to open the debug probe from the given commandline
+        // arguments. This ensures that there is only one probe
+        // connected or if multiple probes are found, a single one is
+        // specified via the commandline parameters.
```

Comment:
Please make a proper doc comment out of this.

Reply:
Removed for now. Will write docs before final review.